core/lib/l1_contract_interface/src/i_executor/methods/commit_batches.rs

```diff
@@ -10,6 +15,7 @@ use crate::{
 pub struct CommitBatches {
     pub last_committed_l1_batch: L1BatchWithMetadata,
     pub l1_batches: Vec<L1BatchWithMetadata>,
+    pub l1_batch_commit_data_generator: Arc<dyn L1BatchCommitDataGenerator>,
```

Comment:
Bikeshedding: IMO, such an interface looks bad. Previously, `CommitBatches` was plain old data, but now it includes behavior. Dumb question: Why is `L1BatchCommitDataGenerator` needed in the first place? Don't its implementations correspond 1-to-1 to `L1BatchCommitDataGeneratorMode`? If so, wouldn't it make sense to pass `L1BatchCommitDataGeneratorMode` instead and attach logic currently present in the `L1BatchCommitDataGenerator` trait to it?

Reply:
This would likely change. The current proposal is as follows (the comment went missing for some reason on this PR, so I shared it to Lambda directly):
![image](https://github.com/matter-labs/zksync-era/assets/12111581/af4ca180-35f6-440c-99d0-12655552e98b)

Reply:
Hello @popzxc , thank you for taking the time to review the PR and provide your suggestions. They all make sense to us too, and we've already filed an issue to track them, along with other suggestions from you and others. While most of them have been addressed, some are significant changes that we believe go beyond the scope of what we've discussed for this initial release.

We intend to revisit these improvements after this PR is merged, as the Validium feature will continue its development after the first delivery. We will not be closing the feature development.